app/databasemanager.py

```diff
@@ -17,16 +28,66 @@ def create_connection(self) -> Connection:
         try:
             return sqlite3.connect(self.db)
         except Error as e:
-            print("Error connecting to database: ", e)
+            print('Error connecting to database: ', e)
 
 
     def close_connection(self):
         """Close the Connection to the database"""
         try:
             self.connection.close()
         except Error as e:
-            print("Error closing connection to database: ", e)
+            print('Error closing connection to database: ', e)
+
+
+    def import_dataset(self, path: str):
+        """Imports dataset specified by path and adds it to self.db.
+        Returns whether data was successfully imported into self.db.
+
+        Assume the dataset is a csv of the form: Country, Country-code, year, prevalence.
+        """
+        try:
+            df = pd.read_csv(path, header=0, usecols=[0,2,3], names=[Table.COLUMN_ENTITY, Table.COLUMN_YEAR, Table.COLUMN_PREVALENCE])
+        except FileNotFoundError:
+            print(f'Could not find valid dataset file from {path}')
+            return False
+        except:
+            print(f'Error reading from {path}')
+            return False
+
+        try:
+            df.to_sql(Table.NAME, self.connection, if_exists='replace', index=False)
+        except:
+            print('Error importing dataset into database')
+            return False
+
+        return True
 
 
     def get_all_data(self):
```

Comment:
maybe instead of printing, return the data so we can use it elsewhere. A good question would be to think about, what shape should the data be? I think you could make a class, where each class represents a row in the table and just return an array of that.

Reply:
Will investigate and discuss